### PR TITLE
Feature: create merkle tree from vector

### PR DESCRIPTION
**Pull request by tomip01**
* Creates the tree out a vector of strings
* Allows the creation with an arbitrary length of input
* Test for valid creation of different length input
* Use sha3 for hashing library

### CODE REVIEW

File: src/merkle.rs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+impl MerkleTree {
+    pub fn new(data: Vec<&str>) -> MerkleTree {
+        let mut merkle = MerkleTree { tree: Vec::new() };
+        if data.is_empty() {
+            return merkle;
+        }
+
+        // push hashes of the input
+        let leafs: Vec<Hash> = data.iter().map(|value| hash(value)).collect();
```
Comment (LucasUTNFRD) [inline src/merkle.rs:28]:
```suggestion
        let leafs: Vec<Hash> = data.iter().map(hash).collect();
```
Here the clousure is not needed

**Review by LucasUTNFRD — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+run:
+	cargo run
+test:
+	cargo test
+check:
+	cargo check
+lint:
+	cargo clippy -- -D warnings
+
+.PHONY: all run test check lint
```
Comment (lima-limon-inc) [inline Makefile:12]: Don't forget the empty blank line at the end ;)

**Review by damiramirez — APPROVED** [review]
Nice work! LGTM

**Review by lima-limon-inc — APPROVED** [review]

File: src/merkle.rs (hunk 1 above)
Comment (tomip01) [reply src/merkle.rs:28]: It doesn't compile :(

Comment (lima-limon-inc) [reply src/merkle.rs:28]: https://www.youtube.com/watch?v=CGj85pVzRJs

**Pull request merged**